plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -240,11 +250,25 @@ func compareTopics(expected []string, incoming []string) bool {
 }
 
 func (m *MQTTConsumer) onMessage(acc telegraf.TrackingAccumulator, msg mqtt.Message) error {
+	m.bytesRecv.Set(0)
```

Comment:
Here you reset the counter on every received message. This shouldn't be done.

Reply:
That wouldn't work. Example: You receive 25 messages the last 10 seconds. 10 seconds is also the interval for the `internal` input, then you will only ever be able to see the size of the last message during that interval..